plugins/defaultplugins/srplugin/vppcalls/srv6_test.go

```diff
@@ -0,0 +1,930 @@
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionX: &srv6.LocalSID_EndX{
+					Psp:               true,
+					NextHop:           nextHop.String(),
+					OutgoingInterface: ifaceBOutOfidxs,
+				},
+			},
+		},
+		{
+			Name:          "invalid IP address (addition with endX behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionX: &srv6.LocalSID_EndX{
+					Psp:               true,
+					NextHop:           invalidIPAddress,
+					OutgoingInterface: ifaceA,
+				},
+			},
+		},
+		{
+			Name:          "missing interface in swIndexes (addition with endDX2 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX2: &srv6.LocalSID_EndDX2{
+					VlanTag:           1,
+					NextHop:           nextHop.String(),
+					OutgoingInterface: ifaceBOutOfidxs,
+				},
+			},
+		},
+		{
+			Name:          "invalid IP address (addition with endDX2 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX2: &srv6.LocalSID_EndDX2{
+					VlanTag:           1,
+					NextHop:           invalidIPAddress,
+					OutgoingInterface: ifaceA,
+				},
+			},
+		},
+		{
+			Name:          "missing interface in swIndexes (addition with endDX4 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX4: &srv6.LocalSID_EndDX4{
+					NextHop:           nextHopIPv4.String(),
+					OutgoingInterface: ifaceBOutOfidxs,
+				},
+			},
+		},
+		{
+			Name:          "invalid IP address (addition with endDX4 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX4: &srv6.LocalSID_EndDX4{
+					NextHop:           invalidIPAddress,
+					OutgoingInterface: ifaceA,
+				},
+			},
+		},
+		{
+			Name:          "rejection of IPv6 addresses (addition with endDX4 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX4: &srv6.LocalSID_EndDX4{
+					NextHop:           nextHop.String(),
+					OutgoingInterface: ifaceA,
+				},
+			},
+		},
+		{
+			Name:          "missing interface in swIndexes (addition with endDX6 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX6: &srv6.LocalSID_EndDX6{
+					NextHop:           nextHop.String(),
+					OutgoingInterface: ifaceBOutOfidxs,
+				},
+			},
+		},
+		{
+			Name:          "invalid IP address (addition with endDX6 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX6: &srv6.LocalSID_EndDX6{
+					NextHop:           invalidIPAddress,
+					OutgoingInterface: ifaceA,
+				},
+			},
+		},
+	}
+
+	// Run all cases
+	for _, td := range cases {
+		t.Run(td.Name, func(t *testing.T) {
+			func() { //wrapping in another function to properly teardown things inside deferred function in case of assertion failure (i.e. connection)
```

Comment:
This shouldn't be needed, since this is already inside function that is parameter for `t.Run(...)`.

Reply:
fixed in 069205aedb4b4662bc275579e0f469d7bcd013b5